core/src/main/java/org/apache/iceberg/MetadataLogsTable.java

```diff
@@ -0,0 +1,116 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.List;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.SnapshotUtil;
+
+public class MetadataLogsTable extends BaseMetadataTable {
+
+  private static final Schema METADATA_LOGS_SCHEMA = new Schema(
+      Types.NestedField.required(1, "timestamp_millis", Types.LongType.get()),
+      Types.NestedField.required(2, "file", Types.StringType.get()),
+      Types.NestedField.optional(3, "latest_snapshot_id", Types.LongType.get()),
+      Types.NestedField.optional(4, "latest_schema_id", Types.IntegerType.get()),
+      Types.NestedField.optional(5, "latest_sequence_number", Types.LongType.get())
+  );
+
+  MetadataLogsTable(TableOperations ops, Table table) {
+    this(ops, table, table.name() + ".metadata_log");
+  }
+
+  MetadataLogsTable(TableOperations ops, Table table, String name) {
+    super(ops, table, name);
+  }
+
+  @Override
+  MetadataTableType metadataTableType() {
+    return MetadataTableType.METADATA_LOGS;
+  }
+
+  @Override
+  public TableScan newScan() {
+    return new MetadataLogScan(operations(), table());
+  }
+
+  @Override
+  public Schema schema() {
+    return METADATA_LOGS_SCHEMA;
+  }
+
+  private DataTask task(TableScan scan) {
+    TableOperations ops = operations();
+    List<TableMetadata.MetadataLogEntry> metadataLogEntries =
+        Lists.newArrayList(ops.current().previousFiles().listIterator());
+    metadataLogEntries.add(new TableMetadata.MetadataLogEntry(ops.current().lastUpdatedMillis(),
+        ops.current().metadataFileLocation()));
+    return StaticDataTask.of(
+        ops.io().newInputFile(ops.current().metadataFileLocation()),
+        schema(),
+        scan.schema(),
+        metadataLogEntries,
+        metadataLogEntry -> MetadataLogsTable.metadataLogsTableToRow(metadataLogEntry, table())
+    );
+  }
+
+  private class MetadataLogScan extends StaticTableScan {
+    MetadataLogScan(TableOperations ops, Table table) {
+      super(ops, table, METADATA_LOGS_SCHEMA, MetadataTableType.METADATA_LOGS, MetadataLogsTable.this::task);
+    }
+
+    MetadataLogScan(TableOperations ops, Table table, TableScanContext context) {
+      super(ops, table, METADATA_LOGS_SCHEMA, MetadataTableType.METADATA_LOGS, MetadataLogsTable.this::task, context);
+    }
+
+    @Override
+    protected TableScan newRefinedScan(TableOperations ops, Table table, Schema schema, TableScanContext context) {
+      return new MetadataLogScan(ops, table, context);
+    }
+
+    @Override
+    public CloseableIterable<FileScanTask> planFiles() {
+      return CloseableIterable.withNoopClose(MetadataLogsTable.this.task(this));
+    }
+  }
+
+  private static StaticDataTask.Row metadataLogsTableToRow(TableMetadata.MetadataLogEntry metadataLogEntry,
```

Comment:
Nit: its not really a whole table to row, can we change to metadataLogToRow?  (

Reply:
makes sense, done !